libvast/vast/detail/line_range.hpp

```diff
@@ -36,6 +38,14 @@ class line_range : range_facade<line_range> {
 
   size_t line_number() const;
 
+  void set_read_timeout(std::chrono::milliseconds timeout);
+
+  void unset_read_timeout();
+
+  // Did the last call to `next()` fail due to a timeout failure?
+  // Can only return true if a timeout was set.
+  bool timed_out() const;
```

Comment:
This API is a little bit too easy to misuse. I'd like to remove this function and add a variant of `next()` instead:
```
[[nodiscard]] bool next_timeout(std::chrono::milliseconds timeout);
```
Internally, I would set the timeout to the fdinbuf only if it is different from the last timeout. We can do so because we own `input_` exclusively.

We can also remove `set_read_timeout()` and `unset_read_timeout()` from the API.

Reply:
I agree, the `next_timeout()` API looks much nicer.

> Internally, I would set the timeout to the fdinbuf only if it is different from the last timeout. We can do so because we own input_ exclusively.

I'm not sure if this is actually an optimization, this will save a write but introduce an additional load and two comparisons. (not that it really matters tbh, because when we hit this path we actually introduce an additional syscall on the next read(), so a few instructions more or less will be undetectable)